fulltext/__main__.py

```diff
@@ -12,17 +13,40 @@
 import fulltext
 
 
+HERE = os.path.abspath(os.path.dirname(__file__))
+
+
 def _handle_open(path):
     with open(path, 'rb') as f:
         return fulltext.get(f)
 
 
-def main(args=sys.argv[1:]):
+def test_backends():
+    """Invoke test() for all backends and fail (raise) if some dep
+    is missing.
     """
-    Extract text from a file.
+    path = os.path.join(HERE, "backends")
+    for name in os.listdir(path):
+        if not name.endswith('.py'):
+            continue
+        mod_name = "fulltext.backends.%s" % (
+            os.path.splitext(os.path.basename(name))[0])
+        mod = __import__(mod_name, fromlist=[' '])
+        print("checking %r" % mod)
+        if hasattr(mod, "check"):
+            mod.check()
```

Comment:
Can you modify this so that output is report-style. The idea is so that I can run this command on our appliance as part of CI/QA process to ensure all tools are present. Having a list of everything missing will help me to resolve the problem(s) quickly.

Be sure to exit with non-zero code if any problems are detected.

Reply:
so you want to print all errors first then finally sys.exit(1)?

Reply:
Yessir.